### PR TITLE
works on exp-4339: Add context menu search telemetry test.

### PR DESCRIPTION
**Pull request by jrbenny35**
Adds the context menu search test.

### CODE REVIEW

File: tests/scenarios/test_withads_search.py [hunk 1]
```diff
@@ -59,11 +67,20 @@ def search_using_search_bar_to_return_ads(selenium):
...
+    selenium.get("http://localhost:8000")
+    el = selenium.find_element(By.CSS_SELECTOR, "#search-to-return-ads")
+
+    ActionChains(selenium).move_to_element(el).pause(1).double_click(el).pause(1).context_click(
+        el
+    ).perform()
+    with selenium.context(selenium.CONTEXT_CHROME):
+        menu = selenium.find_element(By.CSS_SELECTOR, "#contentAreaContextMenu")
+        menu.find_element(By.CSS_SELECTOR, "#context-searchselect").click()
+    WebDriverWait(selenium, 60).until(EC.number_of_windows_to_be(3))
```
Comment (pdehaan) [inline tests/scenarios/test_withads_search.py:81]: Seems interesting&hellip; Is this just counting that a new window has been opened with search results?

File: tests/scenarios/test_withads_search.py [hunk 2]
```diff
@@ -59,11 +67,20 @@ def search_using_search_bar_to_return_ads(selenium):
...
 
 
-@then("The browser reports correct telemetry for the searchbar event")
-def check_telemetry_for_with_ads_search_bar(find_ads_search_telemetry):
-    find_ads_search_telemetry("browser.search.withads.searchbar", ping_data={"google:tagged": 1})
+@then(parsers.parse("The browser reports correct telemetry for the {search:l} search event"))
```
Comment (pdehaan) [inline tests/scenarios/test_withads_search.py:84]: Had to do some googling for this, but verrry nice! 👏 
https://pytest-bdd.readthedocs.io/en/stable/#step-arguments

File: tests/fixtures/index.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Klaatu Test Page</title>
+    </head>
+    <body>
+        <h1>This is a test page for Klaatu tests :)</h1>
+        <h2>Search elements</h2>
+        <ul>
+            <li id="copy-paste-string">Mozilla</li>
```
Comment (pdehaan) [inline tests/fixtures/index.html:11]: not really relevant to anything, but doesn't look like this is used [yet?]

File: tests/features/withads_search.feature [hunk 4]
```diff
@@ -5,10 +5,16 @@ Feature: Search tests that return ads to verify telemetry
...
         Given Firefox is launched enrolled in an Experiment
         And The user searches for something in the search bar that will return ads
-        Then The browser reports correct telemetry for the searchbar event
+        Then The browser reports correct telemetry for the searchbar search event
+
+    @smoke
+    Scenario: Telemetry reports correctly for context menu search events
+        Given Firefox is launched enrolled in an Experiment
+        And The user highlights some text and wants to search for it via the context menu
+        Then The browser reports correct telemetry for the context menu contextmenu search event
```
Comment (pdehaan) [inline tests/features/withads_search.feature:20]: q: does this need "context menu" and "contextmenu"?
Not sure if that affects the parser stuff below, re:
https://github.com/mozilla/klaatu/blob/270fc66f28c110356bdb3fce19e2987f64fb2265/tests/scenarios/test_withads_search.py#L84-L86

File: pyproject.toml [hunk 5]
```diff
@@ -44,6 +44,8 @@ mypy = "^1.8.0"
 isort = "^5.13.2"
 flake8 = "^7.0.0"
 beautifulsoup4 = "^4.12.3"
+pytest-repeat = "^0.9.3"
```
Comment (pdehaan) [inline pyproject.toml:47]: Is this used? It's the only reference to "repeat" I see in this PR, but not sure if its a peer dependency of something maybe. (spoiler: I lack most basic Python knowledge).

**Review by pdehaan — APPROVED** [review]
R+
LGTM

File: tests/scenarios/test_withads_search.py (hunk 1 above)
Comment (jrbenny35) [reply tests/scenarios/test_withads_search.py:81]: That correct. Tabs are "windows" to selenium/geckodriver

File: tests/fixtures/index.html (hunk 3 above)
Comment (jrbenny35) [reply tests/fixtures/index.html:11]: Will be changing the tests that use "http://allizom.org" to use this page as well

File: pyproject.toml (hunk 5 above)
Comment (jrbenny35) [reply pyproject.toml:47]: More so used for testing locally

**Pull request merged**